src/LanguageServer/Protocol/Features/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs

```diff
@@ -14,11 +16,15 @@ private DiagnosticIncrementalAnalyzer CreateIncrementalAnalyzer(Workspace worksp
     private DiagnosticIncrementalAnalyzer CreateIncrementalAnalyzerCallback(Workspace workspace)
     {
         // subscribe to active context changed event for new workspace
-        workspace.DocumentActiveContextChanged += OnDocumentActiveContextChanged;
+        _ = workspace.RegisterDocumentActiveContextChangedHandler(OnDocumentActiveContextChangedAsync);
 
         return new DiagnosticIncrementalAnalyzer(this, AnalyzerInfoCache, this.GlobalOptions);
     }
 
-    private void OnDocumentActiveContextChanged(object? sender, DocumentActiveContextChangedEventArgs e)
-        => RequestDiagnosticRefresh();
+    private Task OnDocumentActiveContextChangedAsync(DocumentActiveContextChangedEventArgs e)
+    {
+        RequestDiagnosticRefresh();
+
+        return Task.CompletedTask;
```

Comment:
this is like the fourth file where it's async,  but completes synchronously.  perhaps we should allow registering synchronous handlers as well?

Reply:
I'm open to the idea, but that would double the number of Register* methods that need to be exposed right? @jasonmalinowski -- any issues with exposing both sync and async supporting registration methods?

Reply:
Switched over to just use synchronous events